### PR TITLE
Automate the release of new versions of the library

### LINKED ISSUE #63: Refactoring component registry
To have components we are in need of with LVN

### PR DESCRIPTION
**Pull request by nelson-glauber**
This PR includes change to automate the release of new versions of the Core Jetpack library. 
The main changes are listed below:
- Migrate the build scripts from Groovy to Kotlin KTS. This change was not required, but [KTS is the new standard](https://android-developers.googleblog.com/2023/04/kotlin-dsl-is-now-default-for-new-gradle-builds.html) for Android development.
- This library is currently hosted in [Jitpack](https://jitpack.io/) which trigger a new build for branches and releases. Therefore, the `jitpack.yml` was updated to include all the necessary steps to configure the build. 
- The `jitpack.yml` allows to include custom commands to customize the build. The `scripts/prepareJitpackEnvironment.sh` file contains a few commands in order to install and configure Rust in order to generate the `*.so` files. 
- Including the "Release a version" section in README file.

### CODE REVIEW

File: jitpack.yml [hunk 1]
```diff
@@ -1,5 +1,24 @@
...
+  - sdk use java 8.0.265-open
+  # Installing the necessary Android SDK packages (accepting licenses, NDK, CMake)
+  - yes | sdkmanager --licenses
+  - yes | sdkmanager --update
+  - yes | sdkmanager --uninstall "ndk-bundle"
+  - yes | sdkmanager --install "ndk;25.1.8937393"
+  - yes | sdkmanager --install "cmake;3.22.1"
+  # Install JDK 17 and using it for the rest of the build
+  - sdk install java 17.0.3-tem
+  - sdk use java 17.0.3-tem
```
Comment (simlay) [inline jitpack.yml:19]: This repo doesn't have dependabot enabled but I'm pretty sure dependabot wouldn't update these which would be nice given how specific the versions are.

File: core/build.gradle.kts [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+    kotlinOptions {
+        jvmTarget = "1.8"
+    }
+    publishing {
+        singleVariant("release") {
+            withSourcesJar()
+            withJavadocJar()
+        }
+    }
+    ndkVersion = "25.1.8937393"
```
Comment (simlay) [inline core/build.gradle.kts:35]: I don't know how semver works in this context (like is this version pinned?) but this seems quite specific.

**Review by simlay — APPROVED** [review]
Assuming this works with the jitpack ci environment, I think this looks good.

File: jitpack.yml (hunk 1 above)
Comment (nelson-glauber) [reply jitpack.yml:19]: The things seemed to be very specific with jitpack... Looking at the logs, at this moment, at the beginning of the pipeline `java 17.0.7-oracle` is being installed. So I thought it wouldn't be necessary to install this one. But turns out it didn't work with the oracle version version... It's weird... :/ 
For future updates, we have to make sure the toolset works fine both locally and in the pipeline...

File: core/build.gradle.kts (hunk 2 above)
Comment (nelson-glauber) [reply core/build.gradle.kts:35]: There's a documented explanation :) This version is the one compatible with Gradle 8.0
https://developer.android.com/studio/projects/install-ndk#default-ndk-per-agp

**Pull request merged**